### PR TITLE
Update create lsp7 token file

### PR DESCRIPTION
**Pull request by skimaharvey**
# What does this PR introduce? 

NB1: **Do not merge this PR until `lsp-smart-contracts` package version `0.14.0` is released as this is using the latest `lsp7` version**

NB2: **Token the liberty to make the CustomToken` more modular so that you do not have to override the contract, but if you think it is too complicated, we can revert to what @kalote did and hardcode the params in the constructor**

NB3: **Please do not merge without testing. I could not test as the new package version is not out as well, and we cannot export the PVT key from the extension atm**

### CODE REVIEW

**Comment by Hugoo** [comment]
Thank you - I converted to draft to make sure we don't merge by accident before the things you mention in the PR are ready.

File: docs/learn/smart-contract-developers/create-lsp7-token.md [hunk 1]
```diff
@@ -14,6 +14,39 @@ You can learn about the project setup by checking the [Getting Started](./gettin
...
+
+### Create a `.env` file
+
+Create a `.env` file in the root of your project and add the following content:
+
+```bash
+PRIVATE_KEY=0x...
+```
+
+Replace the `0x...` with the private key of the account you want to use to deploy your contract or in the case where your are deploying your contract using a Universal Profile, the private key of the EOA that own or has the right permissions on your UP (more information about controllers can be found in the [Key Manager](../../standards/universal-profile/lsp6-key-manager.md) page).
```
Comment (Hugoo) [inline docs/learn/smart-contract-developers/create-lsp7-token.md:48]: This is prob duplicated with the getting started page:

https://docs.lukso.tech/learn/smart-contract-developers/getting-started

We put all the setup in this apge so we don't have to duplicate them in the guides.

image


There is already a warning here so i think you can remove it?

File: docs/learn/smart-contract-developers/create-lsp7-token.md [hunk 2]
```diff
@@ -108,55 +152,78 @@ The `privateKey` coming from your UP extension is the private key of the EOA tha
...
   );
 
-  await tx1.wait();
+  // deploy the Custom Token contract
+  const tx = await universalProfile.execute(
+    1,
+    ethers.ZeroAddress,
+    0,
+    CustomTokenBytecodeWithConstructor,
+  );
```
Comment (Hugoo) [inline docs/learn/smart-contract-developers/create-lsp7-token.md:222]: Maybe to add some context. 
Users reading this guide will prob be very now to the whole LUKSO thing so the more way re route to other docs etc the better.

```suggestion
  // deploy the Custom Token contract
  // https://docs.lukso.tech/contracts/contracts/ERC725/#execute
  const tx = await universalProfile.execute(
    1, // Operation type: CREATE
    ethers.ZeroAddress,
    0, // Value is empty
    CustomTokenBytecodeWithConstructor,
  );
```

File: docs/learn/smart-contract-developers/create-lsp7-token.md (hunk 1 above)
Comment (skimaharvey) [reply docs/learn/smart-contract-developers/create-lsp7-token.md:48]: makes sense I will remove I just want to keep the info about the hardhat version. The problem is that if we add it to the Getting Started page we need to make sure all our guides use the same version. At least `ethers`

File: docs/learn/smart-contract-developers/create-lsp7-token.md [hunk 3]
```diff
@@ -108,55 +125,78 @@ The `privateKey` coming from your UP extension is the private key of the EOA tha
...
-  const provider = new ethers.JsonRpcProvider(
-    'https://rpc.testnet.lukso.network',
-  );
-  // setup signer (the browser extension controller)
-  const signer = new ethers.Wallet(PRIVATE_KEY as string, provider);
   // load the associated UP
-  const UP = await ethers.getContractAt('UniversalProfile', UP_ADDR as string);
-  console.log('🔑 EOA: ', signer.address);
-  console.log('🆙 Universal Profile: ', await UP.getAddress());
+  const universalProfile = await ethers.getContractAtFromArtifact(
```
Comment (CJ42) [inline docs/learn/smart-contract-developers/create-lsp7-token.md:146]: Why was it needed to change the method here from `getContractAt` to `getContractAtFromArtifact`?

**Review by CJ42 — COMMENTED** [review]
added review comments

File: docs/learn/smart-contract-developers/create-lsp7-token.md (hunk 3 above)
Comment (fhildeb) [reply docs/learn/smart-contract-developers/create-lsp7-token.md:146]: > For the following guide, we are usin g... `ethers v6` ...

**Review by fhildeb — APPROVED** [review]

File: docs/learn/smart-contract-developers/create-lsp7-token.md [hunk 4]
```diff
@@ -108,55 +126,78 @@ The `privateKey` coming from your UP extension is the private key of the EOA tha
...
 import { ethers } from 'hardhat';
 import * as dotenv from 'dotenv';
-import UniversalProfile from '@lukso/lsp-smart-contracts/artifacts/LSP0ERC725Account.json';
+import UniversalProfileArtifact from '@lukso/lsp-smart-contracts/artifacts/LSP0ERC725Account.json';
+import { CustomToken__factory } from '../typechain-types';
 
 // load env vars
 dotenv.config();
-const { UP_ADDR, PRIVATE_KEY } = process.env;
+const { UP_ADDR } = process.env;
```
Comment (magalimorin18) [inline docs/learn/smart-contract-developers/create-lsp7-token.md:143]: Is there much value in shortening `UP_ADDRESS` to `UP_ADDR` ?

File: docs/learn/smart-contract-developers/create-lsp7-token.md [hunk 5]
```diff
@@ -108,55 +126,78 @@ The `privateKey` coming from your UP extension is the private key of the EOA tha
...
-    .staticCall(1, ethers.ZeroAddress, 0, CustomTokenBytecode);
+  // add the constructor params to the Custom Token bytecode
+  const CustomTokenBytecodeWithConstructor =
+    CustomTokenBytecode + encodedConstructorParams.slice(2);
+
+  // get the address of the Custom Token contract that will be created
+  const CustomTokenAddress = await universalProfile.staticCall(
+    1, // Operation type: CREATE
+    ethers.ZeroAddress,
+    0, // Value is empty
```
Comment (magalimorin18) [inline docs/learn/smart-contract-developers/create-lsp7-token.md:186]: We could add here if the value is in LYX or in wei

File: docs/learn/smart-contract-developers/create-lsp7-token.md [hunk 6]
```diff
@@ -186,26 +227,31 @@ Deploying with an EOA is more straightforward, but you miss on the Universal Pro
...
 
-  const Token = await customToken.deploy({
-    maxFeePerGas,
-    maxPriorityFeePerGas,
-    type: 2,
-  });
+  // custom token constructor params
+  const tokenName = 'My Custom Token';
+  const tokenSymbol = 'MCT';
+  const tokenContractOwnerAddress = (await ethers.getSigners())[0].address;
```
Comment (magalimorin18) [inline docs/learn/smart-contract-developers/create-lsp7-token.md:237]: can this await fail here ?

also maybe : 
  const tokenContractOwnerAddress = (await ethers.getSigners())[0]?.address;

**Review by magalimorin18 — APPROVED** [review]

**Pull request merged**